### PR TITLE
GH-4997: Enable SparqlBuilder to create VALUES clauses

### PR DESCRIPTION
**Pull request by fkleedorfer**
GitHub issue resolved: #4997 

Briefly describe the changes proposed in this PR:

Adds a builder for VALUES clauses and integrates them with GraphPattern and Select

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Comment by fkleedorfer** [comment]
Happy to rebase that onto develop and change the target branch to develop, if needed.

File: core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/constraint/Values.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
+/* *****************************************************************************
+ * Copyright (c) 2015 Eclipse RDF4J contributors, Aduna, and others.
```
Comment (hmottestad) [inline core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/constraint/Values.java:2]: Could you update this to the current copyright template?

You can find it here: https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md

File: core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/core/query/Query.java [hunk 2]
```diff
@@ -246,7 +256,7 @@ public String getQueryString() {
 		if (offset >= 0) {
 			query.append(OFFSET + " ").append(offset).append("\n");
 		}
-
+		SparqlBuilderUtils.appendAndNewlineIfPresent(values, query);
```
Comment (hmottestad) [inline core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/core/query/Query.java:259]: Doesn't the VALUES clause usually go at the beginning of the WHERE clause?

**Comment by hmottestad** [comment]
> Happy to rebase that onto develop and change the target branch to develop, if needed.

Yeah, this needs to target the develop branch.

Btw. Have you considered supporting VALUES inside UNION clauses? It's a common scenario for getting better performance since the scoping of the UNION means that any VALUES clauses outside the UNION clause will be evaluated after the UNION clauses. Duplicating the VALUES clause inside each branch of the UNION clause fixes the scoping issue so the VALUES clause can be used when executing the joins inside each UNION clause.

**Comment by fkleedorfer** [comment]
You can add it to any GraphPattern, so you can also add it to a Union pattern

**Comment by fkleedorfer** [comment]
... But we might be able to make that more explicit if you like

File: core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/core/query/Query.java (hunk 2 above)
Comment (fkleedorfer) [reply core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/core/query/Query.java:259]: No.  https://www.w3.org/TR/sparql11-query/#sparqlGrammar says:

``` 
[2] | Query  ::=  Prologue
                  ( SelectQuery | ConstructQuery | DescribeQuery | AskQuery )
                  ValuesClause
```

**Pull request merged**